File: day01/main.py
#initialize variables
cal_sum=0
cal_list=[]

#open input file
f=open('input.txt', 'r')
lines = f.readlines()

#do calculation
for line in lines:
    if line.strip() == "":
        cal_list.append(cal_sum)
        cal_sum = 0
    else:
        cal_sum += int(line)
#append last element to list
cal_list.append(cal_sum)
f.close() 
cal_list.sort(reverse=True)

#output
print("part 1:", cal_list[0])
print("part 2:", sum(cal_list[:3]))

File: day01/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

func main() {
	input, _ := os.Open("input.txt")
	defer input.Close()

	scanner := bufio.NewScanner(input)

	cal_sum := 0
	var cal_slice []int

	for scanner.Scan() {
		if scanner.Text() == "" {
			cal_slice = append(cal_slice, cal_sum)
			cal_sum = 0
		} else {
			item, _ := strconv.ParseInt(scanner.Text(), 10, 32)
			cal_sum += int(item)
		}
	}
	cal_slice = append(cal_slice, cal_sum)
	sort.Slice(cal_slice, func(i, j int) bool {
		return cal_slice[i] > cal_slice[j]
	})

	fmt.Println("part 1:", cal_slice[0])

	part2_result := 0
	for i := 0; i < 3; i++ {
		part2_result += cal_slice[i]
	}

	fmt.Println("part 2:", part2_result)

}
